src/test/scala/com/microsoft/hyperspace/index/rules/RuleUtilsTest.scala

```diff
@@ -280,6 +280,62 @@ class RuleUtilsTest extends HyperspaceRuleTestSuite with SQLHelper {
     }
   }
 
+  test(
+    "RuleUtils.getCandidateIndexes: Verify no indexes are matched even if signature matches but " +
+      "hybrid scan is disabled and 'deletedFiles' is non-empty.") {
+    assert(!HyperspaceConf.hybridScanEnabled(spark))
+    // Here's an index where the singature is in sync with the latest data, but "deletedFiles"
+    // list is non-empty.
+    val entry =
+      createIndex("t1iTest", Seq(t1c1), Seq(t1c3), t1ProjectNode, deletedFiles = Seq("f1"))
+
+    // Assert that signature of this new index matches with the latest data.
+    // Below is the logic for signature calculation is picked from RuleUtils class.
+    assert(
+      entry.signature.value.equals(
+        LogicalPlanSignatureProvider
+          .create(entry.signature.provider)
+          .signature(t1ProjectNode)
+          .get))
+
+    val indexManager = IndexCollectionManager(spark)
+    val allIndexes = indexManager.getIndexes(Seq(Constants.States.ACTIVE))
+    assert(allIndexes.exists(_.name.equals("t1iTest")))
+
+    val usableIndexes = RuleUtils.getCandidateIndexes(spark, allIndexes, t1ProjectNode)
+    // Verify that even if signature matched, this index is not picked because of non-empty
+    // "deleted" files.
```

Comment:
To me, this doesn't really validate anything. You should create a valid index log entry, run it thru `getCandidateIndexes` and make sure it returns the index log entry. Create the same but now with `deletedFiles` and make sure nothing returns from `getCandidateIndexes`.

Reply:
thanks @imback82 , udpated tests